vpr/src/draw/draw_floorplanning.cpp

```diff
@@ -100,16 +100,17 @@ static void highlight_partition(ezgl::renderer* g, int partitionID, int alpha) {
     // the on screen units for ezgl to use.
 
     for (int region = 0; (size_t)region < regions.size(); region++) {
-        const auto reg_coord = regions[region].get_region_rect();
+        const vtr::Rect<int>& reg_coord = regions[region].get_region_bounds().get_rect();
+        const auto [layer_begin, layer_end] = regions[region].get_region_bounds().get_layer_range();
 
         //TODO: 0 should be replaced with the actual z value of the region when graph is 3D
-        ezgl::rectangle top_right = draw_coords->get_absolute_clb_bbox(reg_coord.layer_num,
-                                                                       reg_coord.xmax,
-                                                                       reg_coord.ymax,
+        ezgl::rectangle top_right = draw_coords->get_absolute_clb_bbox(layer_begin,
```

Comment:
Should draw what is on the selected layers --> for each region check if it intersects the layer range that is visible.  If so, draw.  If not, don't draw.

Reply:
Or just loop through all visible layers, drawing every region that intersects that layer with the specified transparency. (This is even better I think).